src/geometry/types.rs

```diff
@@ -457,76 +493,76 @@ mod tests {
 
     // Size //////////////////////////////////////////////////////////////////
 
-    #[test]
-    #[should_panic(expected = "A size's width and/or height must be positive.")]
-    fn size_non_positive_width() {
-        // given, when , then
-        Size::new(0, 1);
-    }
+//    #[test]
+//    #[should_panic(expected = "A size's width and/or height must be positive.")]
+//    fn size_non_positive_width() {
+//        // given, when , then
+//        Size::new(2);
+//    }
 
-    #[test]
-    #[should_panic(expected = "A size's width and/or height must be positive.")]
-    fn size_non_positive_height() {
-        // given, when , then
-        Size::new(10, 0);
-    }
+//    #[test]
+//    #[should_panic(expected = "A size's width and/or height must be positive.")]
+//    fn size_non_positive_height() {
+//        // given, when , then
+//        Size::new(2);
+//    }
 
     // Rect //////////////////////////////////////////////////////////////////
 
     #[test]
     fn rect_diameter() {
         // given
-        let sut = Rect::new(0.0, 0.0, 3, 4);
+        let sut = Rect::new(0.0, 0.0, Size::new(2));
 
         // when
         let result = sut.diameter();
 
         // then
-        assert_eq!(5.0, result);
+        assert_eq!(5.65685424949, result);
     }
 
-    #[test]
-    #[should_panic(expected = "A size's width and/or height must be positive.")]
-    fn rect_non_positive_size() {
-        // given, when , then
-        Rect::new(-1.0, 1.0, 1, 0);
-    }
+//    #[test]
+//    #[should_panic(expected = "A size's width and/or height must be positive.")]
+//    fn rect_non_positive_size() {
+//        // given, when , then
+//        Rect::new(-1.0, 1.0, Size::new(2));
+//    }
 
     #[test]
     fn rect_even_quadrants() {
         // given
-        let sut = Rect::new(4.0, 2.0, 6, 8);
+        let sut = Rect::new(4.0, 2.0, Size::new(2));
 
         // when
         let (nw, ne, sw, se) = sut.quadrants();
 
         // then
-        assert_eq!(NW(Rect::new(4.0, 6.0, 3, 4)), nw);
-        assert_eq!(NE(Rect::new(7.0, 6.0, 3, 4)), ne);
-        assert_eq!(SW(Rect::new(4.0, 2.0, 3, 4)), sw);
-        assert_eq!(SE(Rect::new(7.0, 2.0, 3, 4)), se);
+        assert_eq!(NW(Rect::new(4.0, 4.0, Size::new_wh(2,2))), nw);
+        assert_eq!(NE(Rect::new(6.0, 4.0, Size::new_wh(2,2))), ne);
+        assert_eq!(SW(Rect::new(4.0, 2.0, Size::new_wh(2,2))), sw);
+        assert_eq!(SE(Rect::new(6.0, 2.0, Size::new_wh(2,2))), se);
```

Comment:
Can't you use `Size::new(1)` to create a `Size` with width and height 2?

Reply:
yeah you are right , i just forgot to change them, because I used those lines to test